### DISCUSSION TITLE
out variables and using

### DISCUSSION THREAD

**Discussion by jimsnab** [dotnet/csharplang#3798] [Language Ideas]
Can using be added to out?

```
public void MyMethod(string path)
{
    _CreateImage(using out var image);
    image.Save(path, ImageFormat.PNG);
}

void _CreateImage(out Image image)
{
    image = new Image();  // IDisposable
    ... code to draw the image ...
}
```

**Comment by jnm2** [comment]
This is very close to the alternative proposal in #3544, only this is the one we all liked, `using out var`, rather than `dispose out var` or implicitly `out var` which got a lot of downvotes.

**Comment by mpolicki** [comment]
Actually, the alternative proposal in #3544 was to "Introduce a mechanism for the programmer to signal that an `IDisposable` `out` parameter discard should be disposed of", and a `dispose` keyword was merely an example of how that could be done. Also `dispose out var` was not suggested, but `dispose out _` which would signal that the programmer wants to discard the value and call `Dispose` on it.
I wonder if everyone just skimmed that proposal.

**Comment by theunrepentantgeek** [comment]
I don't believe a new issue is needed simply to suggest an alternative syntax for an existing feature suggestion. All that's achieved is to split discussion across two (or more) related issues, making it harder to follow.

**Comment by IS4Code** [comment]
I feel `using` should simply be applicable anywhere a local variable is created. Not just `out using T x` but also `obj is using T x`. I like the possibility that a statement like `_ = obj is using IDisposable _;` would dispose the object automatically whenever applicable.

**Comment by jnm2** [comment]
Duplicate of https://github.com/dotnet/csharplang/discussions/3024
